### PR TITLE
Add optional authentication to MQTT connections

### PR DESCRIPTION
**Pull request by samueltardieu**
MQTT public brokers (such as the one from TheThingsNetwork used with a lot of IoT devices) require a login/password.

### CODE REVIEW

**Comment by samueltardieu** [comment]
Note : failures are unrelated to MQTT

File: mqtt/src/test/scala/MqttSourceSpec.scala [hunk 1]
```diff
@@ -65,6 +68,31 @@ class MqttSourceSpec extends WordSpec with Matchers with ScalaFutures {
...
+      import f._
+
+      val (subscriptionFuture, probe) = MqttSource(p.settings, 8).toMat(TestSink.probe)(Keep.both).run()
+      whenReady(subscriptionFuture.failed) {
+        case e: MqttException => e.getMessage should be("Connection lost")
+        case e                => throw e
+      }
+    }
+
+    "mqtt source" should {
```
Comment (2m) [inline mqtt/src/test/scala/MqttSourceSpec.scala]: Why an additional test nesting?

Comment (samueltardieu) [inline mqtt/src/test/scala/MqttSourceSpec.scala]: What do you mean? It seems to me that this test is similar to others.

Comment (2m) [inline mqtt/src/test/scala/MqttSourceSpec.scala]: You added another `mqtt source` under `mqtt source`. This is the scalatest output of this branch:
```
> mqtt/test
log4j:WARN No appenders could be found for logger (io.moquette.server.config.ConfigurationParser).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
[info] MqttSourceSpec:
[info] mqtt source
[info] - should receive a message from a topic (700 milliseconds)
[info] - should receive messages from multiple topics (243 milliseconds)
[info] - should fail stream when disconnected (454 milliseconds)
[info] - should fail connection when not providing the requested credentials (243 milliseconds)
[info]   should mqtt source
[info]   - should receive a message from a topic with right credentials (243 milliseconds)
[info] - should signal backpressure (251 milliseconds)
[info] - should work with fast downstream (253 milliseconds)
[info] - should support multiple materialization (641 milliseconds)
[info] Run completed in 3 seconds, 133 milliseconds.
[info] Total number of tests run: 8
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 8, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[success] Total time: 3 s, completed Oct 9, 2016 4:33:59 PM
```

**Comment by samueltardieu** [comment]
Again, Cassandra tests are failing, not MQTT ones.

**Comment by samueltardieu** [comment]
retest this please

**Comment by johanandren** [comment]
PLS BUILD

**Pull request merged**